src/field/field2d.cxx

```diff
@@ -45,33 +45,51 @@
 
 #include <bout/assert.hxx>
 
-Field2D::Field2D(Mesh *msh) : fieldmesh(msh), deriv(nullptr) { 
+Field2D::Field2D(Mesh *msh) : Field(msh), deriv(nullptr) { 
 
   boundaryIsSet = false;
 
   if(fieldmesh) {
     nx = fieldmesh->LocalNx;
     ny = fieldmesh->LocalNy;
   }
+#if CHECK > 0
+  else {
+    nx=-1;
+    ny=-1;
+  }
+#endif
   
 #ifdef TRACK
   name = "<F2D>";
 #endif
 }
 
-Field2D::Field2D(const Field2D& f) : fieldmesh(f.fieldmesh), // The mesh containing array sizes
+Field2D::Field2D(const Field2D& f) : Field(f.fieldmesh), // The mesh containing array sizes
                                      data(f.data), // This handles references to the data array
                                      deriv(nullptr) {
+  TRACE("Field2D(Field2D&)");
+  
+#if CHECK > 2
+  checkData(f);
+#endif
+
   if(fieldmesh) {
     nx = fieldmesh->LocalNx;
     ny = fieldmesh->LocalNy;
   }
+#if CHECK > 0
+  else {
+    nx=-1;
+    ny=-1;
+  }
+#endif
   
   boundaryIsSet = false;
-  *this = f;
+  *this = f; //This copies all the data -- unlike 3D which delays, using Copy on write+ref counting
```

Comment:
Doesn't Field2D use the same CoW system as Field3D? The operator= function below uses the same Array reference counting as Field3D